File: Python/pyramid.py
from cs50 import get_int

def main():

    #get pyramid height from user
    while True:
        n = get_int("Height: ")
        if not n < 1 and not n > 8:
            break

    for i in range(n):

        for j in range(n - (i+1)):
            print(" ", end = "")

        for k in range(i+1):
            print("#", end = "")

        print("")

if __name__ == "__main__":
    main()

File: C/pyramid.c
#include <cs50.h>
#include <stdio.h>

int get_pyramid_height(string prompt);
void pyramid(int n);

int main(void)
{
    int i = get_pyramid_height("Height: ");
    pyramid(i);
}

int get_pyramid_height(string prompt)
{
    int n;
    do
    {
       n = get_int("%s", prompt);
    }
    while (n < 1 || n > 8);
    return n;
}

void pyramid(int n)
{
    for (int i = 1; i <= n; i++)
    {
        for (int j = 1; j <= i; j++)
        {
            for (int k = n - 1; k >= i && j == true; k--)
            {
                printf(" ");
            }
            printf("#");
        }
        printf("\n");
    }
}
